main.cpp

```diff
@@ -275,296 +300,283 @@ struct Bakery
         bool burn(int bakingTime);
     };
 
-    //bake bread
-    void bakeBread(double flourAmount, RyeBread brd);
-    //sell cakes
+    void bakeBread(double bFlourAmount, RyeBread brd);
     float cakeSell(std::string cakeName);
-    //smell great
     void smellGreat();
 
     RyeBread breadToSell;
 };
 
-/*
-Thing 3) aqua park
-5 properties:
-    1) number of pools (int)
-    2) number of saunas (int)
-    3) total length of water slides (double)
-    4) opening hours (float)
-    5) ticket price (float)
-3 things it can do:
-    1) relax muscles
-    2) be fun
-    3) learning swimming
- */
+void Bakery::bakeBread(double bFlourAmount, RyeBread brd)
+{
+    std::cout << "Flour used: " << bFlourAmount << std::endl << "Timer set to: " << brd.bakingTime << std::endl;
+}
+
+float Bakery::cakeSell(std::string cakeName)
+{
+    if (cakeName == "Sweet Cake")
+    {
+        std::cout << "Product sold: " << cakeName << std::endl;
+        return 9430.f;
+    }
+    else
+        return 0.f;
+}
+
+void Bakery::smellGreat()
+{
+}
 
 struct AquaPark
 {
-    //number of pools (int)
     int numPools = 5;
-    //number of saunas (int)
     int numSaunas = 3;
-    //total length of water slides (double)
     float totalWaterSlidesLength = 121.56f;
-    //opening hours (float)
     float openAt = 9.30f;
-    //ticket price (float)
     float regularTicker = 13.21f;
 
-    //relax muscles
     void relaxMuscles(bool goToSauna, int swimmingIntensity);
-    //be fun
     void haveFun(float totalWaterSlidesLength, int numberOfSlides);
-    //learning swimming
     bool learnToSwim(int age, int timeSpent);
 };
 
-/*
-Thing 4) house
-5 properties:
-    1) number of windows (int)
-    2) size of living room (float)
-    3) room height (float)
-    4) number of bathrooms (int)
-    5) total floor size (float)
-3 things it can do:
-    1) provide shelter
-    2) provide rest
-    3) get dirty
- */
+void AquaPark::relaxMuscles(bool goToSauna, int swimmingIntensity)
+{
+    goToSauna = false;
+    if (swimmingIntensity > 5)
+        goToSauna = true;
+        std::cout << "You should use one of the sauna's to relax your muscles!" << std::endl;
+}
+void AquaPark::haveFun(float aTotalWaterSlidesLength, int numberOfSlides)
+{
+    std::cout << "Total length of our " << numberOfSlides << " slides is: " << aTotalWaterSlidesLength << " which means that it's not possible to NOT have fun!" << std::endl;
+}
+bool AquaPark::learnToSwim(int age, int timeSpent)
+{
+    if (age < 10 && timeSpent > 60)
+        return true;
+    else if (age >= 10 && timeSpent > 120)
+        return true;
+    else
+        return false;
+
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise